alf/algorithms/oac_algorithm_test.py

```diff
@@ -0,0 +1,141 @@
+# Copyright (c) 2021 Horizon Robotics and ALF Contributors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from absl import logging
+from absl.testing import parameterized
+from functools import partial
+import torch
+import torch.distributions as td
+import unittest
+
+import alf
+from alf.algorithms.config import TrainerConfig
+from alf.algorithms.rl_algorithm import RLAlgorithm
+from alf.algorithms.oac_algorithm import OacAlgorithm, NormalProjectionNetwork
+from alf.algorithms.sac_algorithm import SacState
+from alf.algorithms.rl_algorithm_test import MyEnv
+from alf.data_structures import TimeStep
+from alf.environments.suite_unittest import (PolicyUnittestEnv, ActionType)
+from alf.networks import ActorDistributionNetwork, CriticNetwork
+from alf.utils import common
+from alf.utils.math_ops import clipped_exp
+
+
+class OACAlgorithmTest(parameterized.TestCase, alf.test.TestCase):
+    @parameterized.parameters(True, False)
+    def test_oac_algorithm(self, explore=True):
+        reward_dim = 3
+        num_env = 1
+        config = TrainerConfig(
+            root_dir="dummy",
+            unroll_length=1,
+            unroll_with_grad=True,
```

Comment:
Why do you need unroll_with_grad? I think OAC only changes the exploration but not the training.

Reply:
Same as above, just updated